### ISSUE TITLE
Oracle: Support for Database Link Syntax

### ISSUE THREAD

**Issue by nickbayley** [sqlfluff/sqlfluff#2705]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

It is not possible to use the `oracle` dialect if your SQL makes use of [database links](https://oracle-base.com/articles/misc/managing-database-links).

Before #2608 was implemented, using a database link would cause parsing errors. Afterwards however, it is likely it is parsed successfully, but there might be false positive rule violations because of how it is parsed..

I realise that the above pull request is not associated to database links, but this is what causes the current behaviour.

### Expected Behaviour

For example: 

``` sql
select * from foo@bar where 1 = 1;
```

`foo@bar` would be seen as the table identifier.

Or:

``` sql
select baz.name from foo@bar baz where 1 = 1;
```

`foo@bar` would be the table identifier. `baz` the table alias. And the `where` clause would also be parsed correctly.

### Observed Behaviour

Using the same example: 

``` sql
select * from foo@bar where 1 = 1;
```

Everything that is after and including the `@` is seen as a comment (up until the end of the line).

Here is the output of `sqlfluff parse`:

```
15:18:21 ❯ sqlfluff parse test.sql --dialect oracle
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'select'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    wildcard_expression:
[L:  1, P:  8]      |                        wildcard_identifier:
[L:  1, P:  8]      |                            star:                             '*'
[L:  1, P:  9]      |            whitespace:                                       ' '
[L:  1, P: 10]      |            [META] dedent:
[L:  1, P: 10]      |            from_clause:
[L:  1, P: 10]      |                keyword:                                      'from'
[L:  1, P: 14]      |                whitespace:                                   ' '
[L:  1, P: 15]      |                from_expression:
[L:  1, P: 15]      |                    [META] indent:
[L:  1, P: 15]      |                    from_expression_element:
[L:  1, P: 15]      |                        table_expression:
[L:  1, P: 15]      |                            table_reference:
[L:  1, P: 15]      |                                identifier:                   'foo'
[L:  1, P: 18]      |                    [META] dedent:
[L:  1, P: 18]      |    comment:                                                  '@bar where 1 = 1;'
[L:  1, P: 35]      |    newline:                                                  '\n'
```

### How to reproduce

See one of the examples included above.

### Dialect

`Oracle`

### Version

* `sqlfluff, version 0.10.1`
* `Python 3.10.2`

### Configuration

Didn't make use of any extra configuration, just the defaults.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
